src/main/java/org/aksw/simba/lemming/metrics/metricselection/ComplexityBasedMetricDecider.java

```diff
@@ -0,0 +1,88 @@
+package org.aksw.simba.lemming.metrics.metricselection;
+
+import org.aksw.simba.lemming.ColouredGraph;
+import org.aksw.simba.lemming.metrics.single.SingleValueMetric;
+import org.aksw.simba.lemming.metrics.single.edgetriangles.EdgeNumberOfSimpleTrianglesMetric;
+import org.aksw.simba.lemming.metrics.single.edgetriangles.MultiThreadedNodeNeighborsCommonEdgesMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.EdgeIteratorMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.MultiThreadedNodeNeighborTrianglesMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.NodeIteratorCoreMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.NodeIteratorMetric;
+import org.aksw.simba.lemming.metrics.single.nodetriangles.forward.ForwardMetric;
+
+/**
+ * @author DANISH AHMED on 8/24/2018
+ */
+public class ComplexityBasedMetricDecider {
```

Comment:
Please rename to something like `TriangleMetricsFactory`.

However, it would be **much better** if you could let it simply implement the same interfaces as the triangle metrics and simply create two classes `EdgeTriangleMetric` and `NodeTriangleMetric` which are for every given graph choosing the correct triangle metric instance.